### PR TITLE
Refactor source and dataSourceProvider

### PR DESCRIPTION
**Pull request by gchoqueux**
## Description
- handling fetcher/parser in class `Source.`
- clean up `DataSourceProvider`.

### CODE REVIEW

File: src/Source/Source.js [hunk 1]
```diff
@@ -1,4 +1,26 @@
 import Extent from 'Core/Geographic/Extent';
+import GeoJsonParser from 'Parser/GeoJsonParser';
+import KMLParser from 'Parser/KMLParser';
+import GpxParser from 'Parser/GpxParser';
+import VectorTileParser from 'Parser/VectorTileParser';
+import Fetcher from 'Provider/Fetcher';
+
+export const supportedFetchers = new Map([
```
Comment (zarov) [inline src/Source/Source.js:8]: Note: in the future, we should have these two lists in a separate file, where we can put the format itowns supports by default.

File: src/Source/FileSource.js [hunk 2]
```diff
@@ -136,6 +136,14 @@ class FileSource extends Source {
         return this.url;
     }
 
+    onParsedFile(parsedFile) {
+        this.parsedData = parsedFile;
+        if (parsedFile.extent && parsedFile.extent.crs != 'EPSG:4978') {
```
Comment (zarov) [inline src/Source/FileSource.js:141]: I know it's me that put this filter here, but I don't really remember the case where it is necessary. Do you know if it is still necessary ?

File: src/Provider/DataSourceProvider.js [hunk 3]
```diff
@@ -51,55 +29,33 @@ function parseSourceData(data, extSrc, extDest, layer) {
...
-            }
-        }
-
-        return parsedFile;
-    });
+    return source.parser(data, options).then(parsedFile => source.onParsedFile(parsedFile));
 }
 
-function fetchSourceData(url, layer) {
+function fetchSourceData(extSrc, layer) {
```
Comment (zarov) [inline src/Provider/DataSourceProvider.js:35]: :+1:

File: src/Provider/DataSourceProvider.js [hunk 4]
```diff
@@ -1,40 +1,18 @@
...
 
 function isValidData(data, extentDestination, validFn) {
     if (data && (!validFn || validFn(data, extentDestination))) {
         return data;
     }
 }
 
-const error = (err, url, source) => {
-    source.handlingError(err, url);
+const error = (err, source) => {
```
Comment (zarov) [inline src/Provider/DataSourceProvider.js:9]: Why not replace `url` by `extent` ?

File: src/Provider/DataSourceProvider.js [hunk 5]
```diff
@@ -1,40 +1,18 @@
...
+    source.handlingError(err);
```
Comment (zarov) [inline src/Provider/DataSourceProvider.js:10]: There is a modification to do in `WFSSource` and `test/unit/source.js`

**Review by zarov — APPROVED** [review]
Looks good, if you want to modify some things on the `handlingError` please do, otherwise let's move on.

File: src/Source/FileSource.js (hunk 2 above)
Comment (gchoqueux) [reply src/Source/FileSource.js:141]: because the extent `EPSG:4978` doesn't make sense

File: src/Source/Source.js (hunk 1 above)
Comment (gchoqueux) [reply src/Source/Source.js:8]: yes

File: src/Provider/DataSourceProvider.js (hunk 5 above)
Comment (gchoqueux) [reply src/Provider/DataSourceProvider.js:10]: thanks

File: src/Source/FileSource.js (hunk 2 above)
Comment (zarov) [reply src/Source/FileSource.js:141]: Ah yes that's #1063 my bad

**Pull request merged**